lib/markdown2.py

```diff
@@ -248,6 +248,9 @@ def __init__(self, html4tags=False, tab_width=4, safe_mode=None,
                 self._toc_depth = self.extras["toc"].get("depth", 6)
         self._instance_extras = self.extras.copy()
 
+        if 'link-patterns' in self.extras:
+            if link_patterns is None and self.extras['link-patterns'] is None:
```

Comment:
What's up with the additional  `and self.extras['link-patterns'] is None`? I don't think passing the link patterns is supported through the extras as a dict. Or is this future proofing? Or am I totally missing something :P

Reply:
I assumed that was a supported use case after this was given as the use case in #430 

> ```python
> content = markdown2.markdown( content, extras=["fenced-code-blocks", "tables", "link-patterns"])
> ```

If that is not the case I can remove the extra `and self.extras['link-patterns'] is None`

Reply:
Yup I think you can nuke it, as nothing is passed in through the extras kwarg.